shell/prompt.go

```diff
@@ -12,9 +12,14 @@ import (
 // Prompt the user for text in the CLI. Returns the text entered by the user.
 func PromptUserForInput(prompt string, terragruntOptions *options.TerragruntOptions) (string, error) {
 	// We are writing directly to ErrWriter so the prompt is always visible
-	// no matter what logLevel is configured.
+	// no matter what logLevel is configured. If `--non-interactive` is set, we return "yes" early,
+	// so that prompt is not shown
 	//
 	// See https://github.com/gruntwork-io/terragrunt/issues/1524 for additional context
+	if terragruntOptions.NonInteractive {
+		terragruntOptions.Logger.Infof("The non-interactive flag is set to true, so assuming 'yes' for all prompts")
```

Comment:
Hm, logging this without the prompt itself doesn't make much sense, as the user will have no idea what the "yes" was in response to! It also creates noise in the logs. I think we'd be better off:

1. Check if `--non-interactive` is set
2. If it is, log the prompt and then this "assuming yes" message, both using `Debugf`.
3. If it isn't, write the prompt to `stderr` directly.

Reply:
@brikis98 Updated. I initially thought that displaying the prompt and skipping it with assume message does not make much sense, but from a user stand point it is probably more clear. 👍

Reply:
Roger, thx!